src/imap.rs

```diff
@@ -1411,19 +1418,22 @@ impl Imap {
                         err
                     );
                 }
-            };
-
-            while let Some(Ok(msg)) = msgs.next().await {
+            }
+            .filter_map(|r| async { r.ok() })
+            .collect::<Vec<_>>()
```

Comment:
Decided to get rid of collecting all messages and sorting then. If messages arrive in the requested order, we can dispatch them immediately (and consume much less memory), and if not, just put out-of-order messages to HashMap and take them from there later. It's not too complex

Reply:
The code doesn't look like you got rid of collecting all messages and sorting them?

I agree that we should get rid of the collecting, because it will lead to first fetching all messages and then starting to receive them, which will noticeably increase latency when receiving many messages at once.

Reply:
FTR, we already have this pipelining on master (i.e. emails are fetched and processed in parallel), but this PR removes pipelining. And there were "pipelining efforts" about splitting it more.

Reply:
Decided to do this now rather than get some regression for "sane" servers that don't reorder messages